examples/example_glfw_vulkan/main.cpp

```diff
@@ -68,33 +69,77 @@ static VKAPI_ATTR VkBool32 VKAPI_CALL debug_report(VkDebugReportFlagsEXT flags,
 }
 #endif // IMGUI_VULKAN_DEBUG_REPORT
 
-static void SetupVulkan(const char** extensions, uint32_t extensions_count)
+// Check if the enumration extension has to be enabled
+bool checkExtensions(VkInstance instance)
+{
+    uint32_t extensionCount = 0;
+    vkEnumerateInstanceExtensionProperties(nullptr, &extensionCount, nullptr);
+    std::vector<VkExtensionProperties> availableExtensions(extensionCount);
+    vkEnumerateInstanceExtensionProperties(nullptr, &extensionCount, availableExtensions.data());
+
+    bool enablePortabilitySubset = false;
+    bool portabilitySubsetEnabled = false;
+    bool enableGetPhysicalDeviceProperties2 = false;
+    bool getPhysicalDeviceProperties2Enabled = false;
+
+#if defined(__APPLE__)
+    const bool macOS = true;
+#else
+    const bool macOS = false;
+#endif
+
+    for (const auto &extension : availableExtensions)
+    {
+        if (strcmp(extension.extensionName, VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME) == 0)
+        {
+            portabilitySubsetEnabled = true;
+        }
+        else if (strcmp(extension.extensionName, VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME) == 0)
+        {
+            getPhysicalDeviceProperties2Enabled = true;
+        }
+    }
+
+    if (portabilitySubsetEnabled && macOS)
+    {
+        uint32_t vulkanVersion;
+        vkEnumerateInstanceVersion(&vulkanVersion);
+        if (VK_VERSION_MAJOR(vulkanVersion) >= 1 && VK_VERSION_MINOR(vulkanVersion) >= 3 && VK_VERSION_PATCH(vulkanVersion) >= 216)
```

Comment:
This test would fail for future major or minor version bumps such as 1.4.0 or 2.0.0 where the lesser version components reset to lower numbers.